src/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.js

```diff
@@ -0,0 +1,65 @@
+import {
+  messages,
+  ruleName,
+} from ".."
+import rules from "../../../rules"
+import { testRule } from "../../../testUtils"
+
+const rule = rules[ruleName]
+
+testRule(rule, {
+  ruleName,
+  config: [true],
+
+  accept: [ {
+    code: "a { margin-right: 10px; margin-top: 20px; }",
+  }, {
+    code: "a { margin-right: 10px; margin-top: 20px; margin-bottom: 30px; }",
+  }, {
+    code: "a { padding-left: 10px; margin-right: 10px; margin-top: 20px; margin-bottom: 30px; }",
+  }, {
+    code: "a { padding-top: 1px; padding-bottom: 1px; } b { padding-left: 1px; padding-right: 1px; }",
+  } ],
+
+  reject: [ {
+    code: "a { margin-left: 10px; margin-right: 10px; margin-top: 20px; margin-bottom: 30px; }",
+    message: messages.expected("margin-*"),
```

Comment:
So we end up with a message `Expected shorthand property "margin-*"`? That's wrong, right?